src/software/ai/hl/stp/play/defense_play.cpp

```diff
@@ -70,44 +67,70 @@ void DefensePlay::getNextTactics(TacticCoroutine::push_type &yield, const World
                                   CreaseDefenderAlignment::RIGHT);
         result[0].emplace_back(std::get<1>(crease_defender_tactics));
 
-        // Assign ShadowEnemy tactics until we have every enemy covered. If there any
-        // extra friendly robots, have them perform a reasonable default defensive tactic
-        if (enemy_threats.size() > 0)
+        // Determine how many "immediate" enemy threats there are. If there is only one we
+        // have both shadow enemy tactics swarm and block the "immediate" threat.
+        // Otherwise we assign ShadowEnemy tactics for the next highest threats. If there
+        // any extra friendly robots, have them perform a reasonable default defensive
+        // tactic
+        int immediate_enemy_threats = 0;
+        for (auto enemy_threat : enemy_threats)
         {
-            defense_shadow_enemy_tactic->updateControlParams(enemy_threats.at(1));
-            result[0].emplace_back(defense_shadow_enemy_tactic);
+            if (distance(world.field().friendlyGoal(), enemy_threat.robot.position()) <
+                play_config->getDefensePlayConfig()
+                    ->getImmediateThreatDistance()
+                    ->value())
+            {
+                immediate_enemy_threats++;
+            }
         }
-        else
+
+        if (immediate_enemy_threats == 1)
         {
             auto swarm_ball_tactics = moveRobotsToSwarmEnemyWithBall(move_tactics, world);
             result[0].insert(result[0].end(), swarm_ball_tactics.begin(),
                              swarm_ball_tactics.end());
         }
-
-        if (enemy_threats.size() > 1)
-        {
-            shadow_enemy_tactic->updateControlParams(enemy_threats.at(0),
-                                                     ROBOT_MAX_RADIUS_METERS * 3);
-            result[0].emplace_back(shadow_enemy_tactic);
-        }
         else
         {
-            auto nearest_enemy_robot =
-                world.enemyTeam().getNearestRobot(world.ball().position());
-            if (nearest_enemy_robot)
+            if (enemy_threats.size() > 0)
+            {
+                std::get<0>(shadow_enemy_tactics)
+                    ->updateControlParams(enemy_threats.at(1),
+                                          ROBOT_MAX_RADIUS_METERS * 3);
```

Comment:
looks like ROBOT_MAX_RADIUS_METERS * 3 is being used all over the place, can we pull this out into a constant of defense play?

Reply:
This is probably out of scope because there are a lot of uses of ROBOT_MAX_RADIUS_METERS * some integer across the entire codebase

Reply:
my comment is about the uses within just this defense play, i.e. make a constants in defense_play.h

Reply:
added